executor/executor_bsd.h

```diff
@@ -45,21 +45,45 @@ static intptr_t execute_syscall(const call_t* c, intptr_t a[kMaxArgs])
 {
 	if (c->call)
 		return c->call(a[0], a[1], a[2], a[3], a[4], a[5], a[6], a[7], a[8]);
+#if GOOS_darwin
+// FIXME(HerrSpace): syscall was deprecated on macos:
+// error: 'syscall' is deprecated: first deprecated in macOS 10.12 - syscall(2) is
+// unsupported; please switch to a supported interface. For SYS_kdebug_trace use kdebug_signpost().
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
```

Comment:
Can you please elaborate a bit on your idea? What would the flag represent and what would I do with that flag here? Also note that these are clang pragmas, so we really need to hide them from gcc via the precompiler.

Reply:
There should be something like -Wno-deprecated-declarations compiler flag, no?
sys/targets/targets.go lists compiler flags for an OS. If we add it there, I think it should work. I think compilers should ignore unknown -Wno- flags, or there should be an additional flag that makes compilers ignore them. But we also don't have 2 compilers yet (can XNU be compiled with gcc at all? Since LLVM is an Apple project, I would assume they may rely on it entirely).